packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -416,9 +415,6 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         const attributes = createAttributes(pkg, isRootDataStore, this.disableIsolatedChannels);
         addBlobToSummary(summarizeResult, dataStoreAttributesBlobName, JSON.stringify(attributes));
 
-        // Add GC details to the summary.
-        addBlobToSummary(summarizeResult, gcBlobKey, JSON.stringify(this.summarizerNode.getGCSummaryDetails()));
```

Comment:
Removing this will result in re-generating GC data for the following scenario -
- A new container runtime writes summary with GC blobs at root.
- A summarizer running older versions of container runtime will not find GC data and will re-generate it.

This is a little expensive but it should be okay. Thoughts @vladsud @pleath?

Reply:
We could leave it for a couple of versions but that means the size of summary will increase due to duplicate GC blobs.

Reply:
re-generating GC blobs - when? and also duplicate data in file - forever, or for some time (assuming file gets re-summarized mostly)?

Reply:
If a summarizer running new runtime versions (containing this change) generates summary and an older summarizer loads from this summary, it will have to re-run full GC because it won't find GC blobs in data store summary trees. This will happen the first time, after that it won't do that anymore.

To avoid the above scenario completely, we have to keep the duplicate blobs until telemetry suggests no older versions of runtime are there. After that, we can remove the duplicate blobs.

Reply:
I assume the older summarizer only needs to re-run full GC if GC is needed? So the choice is between temporarily making all new summary trees larger, or temporarily making old GC's slower? If so, I'd vote for the second.

Reply:
Do we have regression tests that exercise the old summarizer + new summary format scenario? Do they confirm the assumption about re-running full GC only once?

Reply:
I like Vlad's idea of shipping the code that can read the new format dark. Once that saturates, we can enable writing the data in new format. Since sweep won't be enabled for a while, we can wait to enable writing data in new format.

Reply:
Unfortunately, we do not have such tests. It may be possible to write a single for this specific purpose. Let me take a look at it.

Reply:
Can we somewhat orchestrate such test via some manual steps? I.e.
1. Do not change snapshot tests - keep old payloads
2. Introduce change dark - it should continue to pass tests just fine with old snapshots (and write data in old format).
3. Switch to new format - only some snapshot tests should fail - only those that generate snapshots, not those that read and process ops.
4. Update snapshots - tests should now pass.

Reply:
Actually, step one should be - no changes in UTs as well (with exception of maybe additions and obvious things where we had bugs).
This kind of approach will give us more confidence that new code is likely interopping with old code fine

Reply:
1. I removed the updates to snapshot tests after disabling writing new format.
2. It works fine as-is.
3. Validated.
4. Validated.

Reply:
I also updated couple of GC end-to-end tests to use the new format while others use the old format. All tests are working fine. I am looking at cheap ways of adding combination of old / new runtime version for the GC tests.